src/core/gs.hpp

```diff
@@ -87,6 +87,12 @@ struct Vertex
 };
 
 class INTC;
+class Point
```

Comment:
Move the Point class above the INTC declaration.

Also, make Point a struct (with all members set to public by default), as it is small and easily accessible.